receiver/filelogreceiver/filelog.go

```diff
@@ -43,22 +42,45 @@ func (f ReceiverType) Type() configmodels.Type {
 
 // CreateDefaultConfig creates a config with type and version
 func (f ReceiverType) CreateDefaultConfig() configmodels.Receiver {
+	return createDefaultConfig()
+}
+func createDefaultConfig() *FileLogConfig {
 	return &FileLogConfig{
-		ReceiverSettings: configmodels.ReceiverSettings{
-			TypeVal: configmodels.Type(typeStr),
-			NameVal: typeStr,
+		BaseConfig: stanzareceiver.BaseConfig{
+			ReceiverSettings: configmodels.ReceiverSettings{
+				TypeVal: configmodels.Type(typeStr),
+				NameVal: typeStr,
+			},
+			Parsers: stanzareceiver.ParserConfigs{},
+		},
+		Input: stanzareceiver.InputConfig{
+			"type": "file_input",
 		},
 	}
 }
 
-// Decode unmarshals configuration into a log parsing pipeline
-func (f ReceiverType) Decode(cfg configmodels.Receiver) (pipeline.Config, error) {
-	logConfig := cfg.(*FileLogConfig)
-	return stanzareceiver.DecodeOperators(logConfig.Operators)
+// BaseConfig gets the base config from config, for now
+func (f ReceiverType) BaseConfig(cfg configmodels.Receiver) stanzareceiver.BaseConfig {
+	return cfg.(*FileLogConfig).BaseConfig
 }
 
 // FileLogConfig defines configuration for the filelog receiver
 type FileLogConfig struct {
-	configmodels.ReceiverSettings `mapstructure:",squash"`
-	Operators                     stanzareceiver.OperatorConfig `mapstructure:"operators"`
+	stanzareceiver.BaseConfig `mapstructure:",squash"`
+	Input                     stanzareceiver.InputConfig `mapstructure:",remain"`
+}
+
+// DecodeInputConfig unmarshals the input operator
+func (f ReceiverType) DecodeInputConfig(cfg configmodels.Receiver) (*operator.Config, error) {
```

Comment:
The reason this exists is because  stanzareceiver.InputConfig is a generic map of any key/values, right? If we used strongly typed fields would this function become superfluous?

Reply:
Correct. I don't believe we'll need this method at all.